README.rst

```diff
@@ -0,0 +1,325 @@
+
+  ./pom.xml
+  ./src/main/java/co/cdap/guides/workflow/EmailNotifier.java
+  ./src/main/java/co/cdap/guides/workflow/Purchase.java
+  ./src/main/java/co/cdap/guides/workflow/PurchaseCounterByProduct.java
+  ./src/main/java/co/cdap/guides/workflow/PurchaseCounterByUser.java
+  ./src/main/java/co/cdap/guides/workflow/PurchaseEventParser.java
+  ./src/main/java/co/cdap/guides/workflow/PurchaseEventVerifier.java
+  ./src/main/java/co/cdap/guides/workflow/PurchaseResultService.java
+  ./src/main/java/co/cdap/guides/workflow/PurchaseWorkflow.java
+  ./src/main/java/co/cdap/guides/workflow/PurchaseWorkflowApp.java
+
+The CDAP application is identified by the ``PurchaseWorkflowApp`` class. This
+class extends an `AbstractApplication
+<http://docs.cdap.io/cdap/current/en/reference-manual/javadocs/co/cask/cdap/api/app/AbstractApplication.html>`__,
+and overrides the ``configure()`` method to define all of the application components:
+
+.. code:: java
+
+  public class PurchaseWorkflowApp extends AbstractApplication {
+    @Override
+    public void configure() {
+      setName("PurchaseWorkflowApp");
+      setDescription("Application describing the Workflow");
+
+      addStream(new Stream("purchaseEvents"));
+
+      addMapReduce(new PurchaseEventParser());
+      addMapReduce(new PurchaseCounterByUser());
+      addMapReduce(new PurchaseCounterByProduct());
+      addWorkflow(new PurchaseWorkflow());
+
+      scheduleWorkflow(Schedules.createTimeSchedule("HourlySchedule", "Schedule execution every 1 hour", "0 * * * *"),
+                       "PurchaseWorkflow");
+
+      addService(new PurchaseResultService());
+
+      createDataset("purchaseRecords", KeyValueTable.class);
+      createDataset("userPurchases", KeyValueTable.class);
+      createDataset("productPurchases", KeyValueTable.class);
+    }
+  }
+  
+The ``PurchaseWorkflowApp`` application defines a new `Stream
+<http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/streams.html>`__
+where purchase events are ingested. Once the data is
+ingested, the events can be processed in realtime or batch. In our
+application, we will process the events in batch using the
+``PurchaseWorkflow`` program and compute the total purchases made by the particular user
+and total purchases made for the particular product in a specific time range. We will use three MapReduce
+programs ``PurchaseEventParser``, ``PurchaseCounterByUser``, and ``PurchaseCounterByProduct`` to apply
+different processing on the purchase events and Workflow ``PurchaseWorkflow`` to connect these MapReduce
+programs in a desired fashion.
+
+The result of the Workflow execution is persisted into a Dataset; the
+application uses the ``createDataset`` method to define the Dataset. We use three datasets,
+``purchaseRecords`` to store the valid parsed purchase events, ``userPurchases`` to store the total purchases
+made by particular user, and ``productPurchases`` to store the total purchases made for per product.
```

Comment:
made by a particular user; and ``productPurchases`` to store the total purchases made for each product.